code/controllers/subsystem/enumeration.dm

```diff
@@ -0,0 +1,41 @@
+/**
+ * Subsystem that allows for performing operations over
+ * enumerations that involve tick checking with no risk of freezing the MC
+ * as a result of TICK_CHECK waking up before the MC can wake up.
+ *
+ * @PowerfulBacon
+ * */
+SUBSYSTEM_DEF(enumeration)
+	name = "Enumeration"
+	wait = 1
+	flags = SS_NO_INIT | SS_TICKER
+
+	/// Enumerators we are tick check enumerating over
+	var/list/datum/enumerator/tick_enumerations = list()
```

Comment:
redundant type specification 
also doesn't specifying type on list cause undefined behaviors?

Reply:
It doesn't actually do anything else, it just appears that way. The latest version of byond was going to remove it but then they reverted that and kept typed lists.

Reply:
it's actually better because sdmm will do typechecks on it i think